src/test/enginebuffertest.cpp

```diff
@@ -43,6 +45,23 @@ TEST_F(EngineBufferTest, DisableKeylockResetsPitch) {
     ASSERT_EQ(0.0, ControlObject::get(ConfigKey(m_sGroup1, "pitch")));
 }
 
+TEST_F(EngineBufferTest, DisableKeylockKeepsPitch) {
+    // Pitch must not change when unlocking with KeyunlockMode::KeepLockedKey.
+    ControlObject::set(ConfigKey(m_sGroup1, "keylockMode"),
+            1.0); // KeylockMode::LockCurrentKey
+    ControlObject::set(ConfigKey(m_sGroup1, "keyunlockMode"),
+            1.0); // KeyunlockMode::KeepLockedKey
+    ControlObject::set(ConfigKey(m_sGroup1, "file_bpm"), 128.0);
+    ControlObject::set(ConfigKey(m_sGroup1, "keylock"), 1.0);
+    ControlObject::set(ConfigKey(m_sGroup1, "pitch"), 0.5);
+    ProcessBuffer();
+
+    ControlObject::set(ConfigKey(m_sGroup1, "keylock"), 0.0);
+    // We require a buffer process to see that the keylock state has changed.
+    ProcessBuffer();
+    ASSERT_EQ(0.5, ControlObject::get(ConfigKey(m_sGroup1, "pitch")));
```

Comment:
```suggestion
    ASSERT_DOUBLE_EQ(0.5, ControlObject::get(ConfigKey(m_sGroup1, "pitch")));
```

Reply:
0.5 can be represented unambiguously though. So there is a rounding/precision issue somewhere that introduces a slight difference... possibly a bug?

Reply:
There definitely is a round-trip rounding issue, see https://github.com/mixxxdj/mixxx/blob/53f3b073bf19d0af7e745db565b4d1bcf2df6d3e/src/engine/controls/keycontrol.cpp#L276-L281
and #12027 but fixing that is out of scope for this PR IMO.
I suspect it's `KeyControl::updateKeyCOs` recalculating all values, even pitch if the called by `updatePitch()` which is called by `slotPitchChanged(double pitch)`.

I pushed an experimental commit, let's see if that's sufficient.

Reply:
> I pushed an experimental commit, let's see if that's sufficient.

No, it's not.
Can we use NEAR or similiar?

Reply:
probably, that's what `TrackLoadResetsPitch` already does anyways.

https://github.com/mixxxdj/mixxx/blob/0119d85562a81976c3c7678d58cecfae4d0adc5a/src/test/enginebuffertest.cpp#L52-L56

Reply:
:facepalm: